client/src/js/LeftAndMain_Subsites.js

```diff
@@ -32,11 +22,15 @@
     });
 
     /*
-     * Reload subsites dropdown when subsites are added or names are modified
+     * Reload subsites dropdown when subsites are added or modified
      */
-    $('.cms-container .tab.subsite-model').entwine({
-      onadd(e) {
-        $('.cms-container').loadFragment('admin/subsite_xhr', 'SubsiteList');
+    $('#Form_ItemEditForm').entwine({
+      onaftersubmitform(e) {
+        // Only load the fragment if this form is the subsite form.
+        // We can't add a selector to the form itself so check for the tab we added a specific class to.
+        if (this.find('[role="tab"].subsite-model').length > 0) {
+          $('.cms-container').loadFragment('admin/subsite_xhr', 'SubsiteList');
+        }
         this._super(e);
       }
```

Comment:
I think this was trying to reload the selector any time it saw the root.main tab in the subsite edit form.
The selector was wrong, but the approach was wrong too - we should only reload the selector after the form is submitted, not just any time the form loads.